returnn/hdf.py

```diff
@@ -346,6 +364,297 @@ def run(self):
+                "-ar",
+                str(self.target_sampling_rate),
+                "-f",
+                "wav",
+                "-",
+            ],
+            check=True,
+            stdout=sp.PIPE,
+        )
+
+        # we don't read with soundfile as the wave module is more robust
+        with wave.open(io.BytesIO(ffmpeg_proc.stdout), "rb") as audio:
+            assert audio.getnchannels() == 1
+            assert audio.getsampwidth() == 2
+            assert audio.getframerate() == self.target_sampling_rate
+
+            audio_data: bytes = audio.readframes(audio.getnframes())
+
+        audio_data_int16 = array.array("h")
+        audio_data_int16.frombytes(audio_data)
+        if sys.byteorder == "big":
+            audio_data_int16.byteswap()
+        audio_data = np.array(audio_data_int16, dtype=np.int16)
+        if self.output_dtype.startswith("float"):
+            # scale to output range
+            audio_data = audio_data.astype(self.output_dtype) / np.iinfo(audio_data.dtype).max
+
+        def _process_segment(segment: Tuple[str, float, float]):
+            segment_name, start, end = segment
+            if self.rounding == BlissToPcmHDFJob.RoundingScheme.start_and_duration:
+                start = int(start * self.target_sampling_rate / self.round_factor) * self.round_factor
+                duration = int((end - start) * self.target_sampling_rate / self.round_factor) * self.round_factor
+            elif self.rounding == BlissToPcmHDFJob.RoundingScheme.rasr_compatible:
+                start = math.ceil(start * self.target_sampling_rate / self.round_factor) * self.round_factor
+                duration = math.floor(end * self.target_sampling_rate / self.round_factor) * self.round_factor - start
+            else:
+                raise NotImplementedError(f"RoundingScheme {self.rounding} not implemented.")
+            assert start + duration <= len(audio_data)
+            data = audio_data[start : start + duration]
+
+            if self.compress is not None:
+                c_format, c_codec, c_level = self.compress
+                data_arr = array.array("h")
+                data_arr.frombytes(data.tobytes())
+                if sys.byteorder == "big":
+                    data_arr.byteswap()
+                data_bytes = data_arr.tobytes()
+
+                ffmpeg_proc = sp.run(
+                    [
+                        "ffmpeg",
+                        "-hide_banner",
+                        "-loglevel",
+                        "error",
+                        "-y",
+                        "-ar",
+                        str(self.target_sampling_rate),
+                        "-f",
+                        "s16le",
+                        "-i",
+                        "-",
+                        "-c:a",
+                        c_codec,
+                        "-qscale:a",
+                        str(c_level),
+                        "-f",
+                        c_format,
```

Comment:
I think this is too specific, and too less generic, and too complicated.
This `qscale` is codec specific. And often, you would not use this.
E.g., for mp3, I saw that you would maybe use `-c:a libmp3lame -b:a 128k` instead (no `-qscale:a`, also no `-f`). Or for Ogg/Vorbis, `-c:a libvorbis -b:a 64k`.
Or maybe AAC, with VBR: `-c:a libfdk_aac -vbr 3`.

So, instead of this here, I would just put some `*ffmpeg_output_args`.
```suggestion
                        *self.ffmpeg_output_args,
```

Reply:
Oh, yeah. Good point. Implemented.